### DISCUSSION TITLE
The milvus-standalone failed to start on the ARM host

### DISCUSSION THREAD

**Discussion by gaozh424** [milvus-io/milvus#27807] [Q&A and General discussion] [answered]
I attempted to start milvus-standalone v2.3.1 on an ARM-based host running UOS operating system, but it failed to start. Here are the corresponding logs:

milvus-minio       | Warning: The standard parity is set to 0. This can lead to data loss.
milvus-standalone  | <jemalloc>: Unsupported system page size
milvus-standalone  | <jemalloc>: Unsupported system page size
milvus-standalone  | <jemalloc>: Unsupported system page size
milvus-standalone  | [FATAL tini (1)] Failed to allocate memory for child args: 'Cannot allocate memory'

Has anyone encountered this issue before? Are there any suggestions or solutions? Thanks~

**Comment by gaozh424** [comment]
BTW, I used the command "getconf PAGE_SIZE" to query the pageSize, and the result was 65536.

**Comment by yhmo** [comment]
Related with jemalloc.
Wait @jiaoew1991 comment.

**Reply by jiaoew1991** [reply]
It is quite same as this one https://github.com/milvus-io/milvus/issues/27726

**Reply by gaozh424** [reply]
Hi, thank you very much for your response.

So, I have to recompile Milvus on my ARM host?

Is there a pre-compiled image that supports a page size of 64k?

Or are there any tutorials for building ready-to-use Milvus images?

**Reply by jiaoew1991** [reply]
We do not have a pre-compiled image, so we can only recompile Milvus.

https://github.com/milvus-io/milvus/blob/84d05b939bedff778aa688062451819cbf17afa1/internal/core/thirdparty/jemalloc/CMakeLists.txt#L40-L44

change this block of code into 

```
list(APPEND JEMALLOC_CONFIGURE_COMMAND "--with-lg-page=16") 
```

**Reply by xiaofan-luan** [reply]
can we change the config by default?

**Reply by gaozh424** [reply, marked as answer]
Hi~ I finally solved this problem by modifying and recompiling the operating system kernel and changing the PAGE_SIZE parameter to 4K. Now it seems to be running well. This solution is relatively simple and requires less effort. Other friends who encounter this problem can also refer to it. Thank you all for your help~
